isis/src/base/apps/ringsautomos/main.cpp

```diff
@@ -21,104 +23,51 @@ map <QString, void *> GuiHelpers() {
-  //   throw IException(IException::User, msg, _FILEINFO_);
-  // }
-
-  fstream os;
-  bool olistFlag = false;
-  if (ui.WasEntered("TOLIST")){
-    QString olist = ui.GetFileName("TOLIST");
-    olistFlag = true;
-    os.open(olist.toLatin1().data(), std::ios::out);
+  try {
+    ringsautomos(ui, &appLog);
   }
-
-  ProcessMapMosaic m;
-
-  // Set the create flag-mosaic is always created in ringsautomos
-  m.SetCreateFlag(true);
-
-  // Get the Track Flag
-  bool bTrack = ui.GetBoolean("TRACK");
-  m.SetTrackFlag(bTrack);
-
-  ProcessMosaic::ImageOverlay overlay = ProcessMosaic::StringToOverlay(
-      ui.GetString("PRIORITY"));
-
-  if (overlay == ProcessMapMosaic::UseBandPlacementCriteria) {
-    if(ui.GetString("TYPE") == "BANDNUMBER") {
-      m.SetBandNumber(ui.GetInteger("NUMBER"));
-    }
-    else {
-      // Key name & value
-      m.SetBandKeyword(ui.GetString("KEYNAME"), ui.GetString("KEYVALUE"));
+  catch (...) {
+    for (auto grpIt = appLog.beginGroup(); grpIt!= appLog.endGroup(); grpIt++) {
+      Application::Log(*grpIt);
     }
-    // Band Criteria
-    m.SetBandUseMaxValue( (ui.GetString("CRITERIA") == "GREATER") );
+    throw;
   }
-
-  // Priority
-  m.SetImageOverlay(overlay);
-
-  CubeAttributeOutput &oAtt = ui.GetOutputAttribute("MOSAIC");
-  if(ui.GetString("GRANGE") == "USER") {
-    m.RingsSetOutputCube(list,
-                    ui.GetDouble("MINRINGRAD"), ui.GetDouble("MAXRINGRAD"),
-                    ui.GetDouble("MINRINGLON"), ui.GetDouble("MAXRINGLON"),
-                    oAtt, ui.GetFileName("MOSAIC"));
+ 
+  for (auto grpIt = appLog.beginGroup(); grpIt!= appLog.endGroup(); grpIt++) {
+    Application::Log(*grpIt);
   }
-  else {
-    m.RingsSetOutputCube(list, oAtt, ui.GetFileName("MOSAIC"));
-  }
-
-  m.SetHighSaturationFlag(ui.GetBoolean("HIGHSATURATION"));
-  m.SetLowSaturationFlag(ui.GetBoolean("LOWSATURATION"));
-  m.SetNullFlag(ui.GetBoolean("NULL"));
-
-  // Loop for each input file and place it in the output mosaic
-
-  m.SetBandBinMatch(ui.GetBoolean("MATCHBANDBIN"));
+}
 
-  // Get the MatchDEM Flag
-  m.SetMatchDEM(ui.GetBoolean("MATCHDEM"));
 
-  bool mosaicCreated = false;
-  for (int i = 0; i < list.size(); i++) {
-    if (!m.StartProcess(list[i].toString())) {
-      PvlGroup outsiders("Outside");
-      outsiders += PvlKeyword("File", list[i].toString());
-      Application::Log(outsiders);
-    }
-    else {
-      mosaicCreated = true;
-      if(olistFlag) {
-        os << list[i].toString() << endl;
-      }
-    }
-    if(mosaicCreated) {
-      // Mosaic is already created, use the existing mosaic
-      m.SetCreateFlag(false);
-    }
-  }
-  // Logs the input file location in the mosaic
-  for (int i = 0; i < m.imagePositions().groups(); i++) {
-    Application::Log(m.imagePositions().group(i));
-  }
+// Helper function to run calcRange function.
+void helperButtonCalcRange() {
+  UserInterface &ui = Application::GetUserInterface();
```

Comment:
Not sure how much it matters but we may want to pass the UI here

Reply:
same as the other, not even sure if it's possible.